### PR TITLE
DEM-1176 Add Scroll to Form Sections

### PR DESCRIPTION
**Pull request by samsep**
ping @appirio-tech/demandside

### CODE REVIEW

File: src/scripts/controllers/type.controller.coffee [hunk 1]
```diff
@@ -45,6 +45,10 @@ SubmitWorkTypeController = ($scope, SubmitWorkAPIService) ->
 
       resource.$promise.catch (response) ->
 
+  vm.scrollToElement = (elementId) ->
```
Comment (vietnogi) [inline src/scripts/controllers/type.controller.coffee]: angular scroll might have a directive to allow scrolling w.out having to put code in the controller

Comment (samsep) [inline src/scripts/controllers/type.controller.coffee]: The directive seems to only work for anchor tags, in their docs they use code in the controller for buttons

Comment (vietnogi) [inline src/scripts/controllers/type.controller.coffee]: can we create our own directive for buttons?

Comment (samsep) [inline src/scripts/controllers/type.controller.coffee]: Yeah added a directive

File: src/scripts/directives/scroll.directive.coffee [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+directive = ($document) ->
```
Comment (vietnogi) [inline src/scripts/directives/scroll.directive.coffee]: put the directive in ui components library?

**Comment by vietnogi** [comment]
lgtm

**Pull request merged**